src/elm/query.ts

```diff
@@ -237,8 +237,33 @@ export class Query extends Expression {
   }
 
   exec(ctx: Context) {
+    let sourceResults = this.sources.map(s => s.expression.execute(ctx));
+    // If every source is null, the result is null
+    // See: https://jira.hl7.org/browse/FHIR-40225
+    if (sourceResults.every(r => r == null)) {
+      return null;
+    }
+    const isList = sourceResults.some(r => Array.isArray(r));
+    // For ease of processing, convert all the sources to lists and
+    // convert nulls to empty lists (we already handled the ALL null case)
+    sourceResults = sourceResults.map(r => {
+      if (r == null) {
+        return [];
+      } else if (Array.isArray(r)) {
```

Comment:
Same as above. Not sure if this gives us any added benefit or not except consistency.

Reply:
Nah.